tsl/src/nodes/vector_agg/hashing/hash_strategy_common.c

```diff
@@ -8,18 +8,22 @@
 
 #include "nodes/vector_agg/exec.h"
 #include "nodes/vector_agg/grouping_policy_hash.h"
+#include "nodes/vector_agg/vector_slot.h"
 
 /*
  * Allocate enough storage for keys, given that each row of the new compressed
  * batch might turn out to be a new grouping key. We do this separately to avoid
  * allocations in the hot loop that fills the hash table.
  */
 void
-hash_strategy_output_key_alloc(GroupingPolicyHash *policy, DecompressBatchState *batch_state)
+hash_strategy_output_key_alloc(GroupingPolicyHash *policy, TupleTableSlot *vector_slot)
```

Comment:
Only `n` is needed, so let's switch to passing it as argument here?

Reply:
Right, the prepare_for_batch is going to be used for grouping by dictionary-encoded text column later: https://github.com/timescale/timescaledb/pull/7316/files#diff-73404512584d8d432d04eb402e29c1fed22fd787a9e1894e011823f849885bdaR142

Reply:
Ok, I'll just do the changes here in key_alloc then